### ISSUE TITLE
MapDownloader throws Internal Error when different Map Versions installed

### ISSUE THREAD

**Issue by StefanTiess** [heremaps/here-sdk-examples#185]
Hi, there is a nasty bug in the MapLoader for OfflineMaps. It seems linked to the different Map Versions when you download a offlineMap vs after the update that is available immediately.

Steps to reproduce with SDK 4.12.8
* Download an Offline-Map
* Checking for an update will tell you immediately that there is an update available
* Perform the Update until it is complete.
* Install a second Offline-Map.
* Now restart the App without performing the Update for the second Map
* => PersistenMapStatus is OK
* **=> `MapDownloader.getInstalledRegions` throws Exception INTERNAL_ERROR**
* No more MapDownloads or MapUpdates are possible. Every operation will throw INTERNAL_ERROR.
* The Only way to resolve this so far is to call `MapDownloader.clearPersistenMapStorage` which will delete Gigabytes worth of Map Data which is very frustrating for the Driver who needs to redownload everything and me who has to pay DataIO.

Expected would be that the MapLoader can handle different Map Versions and that newly downloaded Maps do not Require an immediate Update.

**Comment by samvgorode** [comment]
Hi, @StefanTiess ! 
I am not from HERE team, but also was curious why should we do immediate update.
Answer is here https://github.com/heremaps/here-sdk-examples/issues/168#issuecomment-1237402843

**Comment by samvgorode** [comment]
@StefanTiess, 
there is a case when interrupted map update causes "INTERNAL_ERROR" all the time. 
When update started and have some percentage - just kill the app. 
I've opened some time ago another issue https://github.com/heremaps/here-sdk-examples/issues/167
They recommend to `mapDownloader.repairPersistentMap` but it will never work.

**Comment by StefanTiess** [comment]
After some investigation I found the issue to be reproducible in the OfflineMaps Example App:
I manipulated it by allowoing the download of multiple regions
```
 String swizNameInGerman = "Schweiz";
        try {
            List<InstalledRegion> list = mapDownloader.getInstalledRegions();
            if(list.size() > 0) swizNameInGerman = "Liechtenstein";
            if(list.size() > 1) swizNameInGerman = "Vatikanstadt";
            if(list.size() > 2) swizNameInGerman = "Monaco";

        } catch (MapLoaderException e) {
            e.printStackTrace();
        }
```

If you kill the app after each downloaded map, reopen it, then try to download the next map, you will get the INTERNAL_ERROR.

Curiously the log shows the follwing warnings:
```
[INFO ] CatalogManager - Existing catalog found, hrn='hrn:here:data::olp-here:ocm', version=56, catalog_handle=0
[WARN ] JournalAlgorithms - IsConfigurationValid issue, dependency not found, hrn='hrn:here:data::olp-here:ocm', dependency='hrn:here:data::olp-here:ocm-japan'
WARN ] CacheUpdateManager - PrefetchMapRegions: journal configuration do not match linked catalogs configuration, catalog_handle=0
 [WARN ] CacheUpdateManager - PrefetchMapRegions: could not start map prefetch job, catalog_handle=0, prefetch_error=kBadRequest, is_cancelled=0
```
Please provide a fix asap or advice for a workaround. I see this issue causing a lot of problems for drivers on the road.

**Comment by samvgorode** [comment]
@datasun , please take a look. This is really important.

**Comment by datasun** [comment]
We have created a ticket on our end to investigate this: #15196.

**Comment by datasun** [comment]
Our team reports that this was fixed with 4.12.11.0. Closing.
